### PR TITLE
logger added

### PR DESCRIPTION
**Pull request by MikhailBerezhanov**
Closes: #5

### CODE REVIEW

**Comment by MikhailBerezhanov** [comment]
Main usage is :

```C
LOG_F_DEBUG(message) 
LOG_F_INFO(message) 
LOG_F_WARNING(message) 
LOG_F_ERROR(message) 
```

Output format example:

```
0711-16:39:48.599 7F5C69FB6000 D Starting ./Linux/test_game
0711-16:39:48.621 7F5C69FB6000 I Log finished.
```

Usage with classes:

```C
class MyCLass : private Logable
{
    MyCLass() : Logable("MyClassName") {}
}
```

That will create prefix logger with `MyClassName` tag before any message from this class .

**Review by Drone29 — CHANGES_REQUESTED** [review]
see comments on the parts needing attention

File: src/main.cpp [hunk 1]
```diff
@@ -5,28 +5,23 @@
...
 // Screen dimension constants
 const int SCREEN_WIDTH = 1000;
 const int SCREEN_HEIGHT = 1000;
 
-int main(int argc, char *args[])
+using namespace GameEngine;
+
+int main(int argc, char *argv[])
 {
+    const LoggerInitializer loggerInitialer(LogLevel::Debug);
```
[the remark this answers is not in the document]
Comment (MikhailBerezhanov) [reply src/main.cpp:18]: This is just an example of usage

File: src/logger.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    private:
+        const LogLevel m_level;
+        ILogger& m_Logger;
+    };
+} // namespace
+
+const std::shared_ptr<GameEngine::ILogger>& GetLogger() noexcept;
+
+#define LOG_DEBUG(message) GetLogger()->Log(GameEngine::LogLevel::Debug, message)
```
[the remark this answers is not in the document]
Comment (MikhailBerezhanov) [reply src/logger.h:103]: Another level of error-handling with  _func_  will be added in next review .

**Comment by MikhailBerezhanov** [comment]
Rebase was needed -> opened new branch

**Pull request closed without merging**